tee-worker/app-libs/stf/src/trusted_call_litentry.rs

```diff
@@ -34,23 +37,84 @@ use litentry_primitives::{
 	Web3Network,
 };
 use log::*;
-use std::vec::Vec;
+use std::{sync::Arc, vec::Vec};
 
 impl TrustedCallSigned {
 	pub fn set_user_shielding_key_internal(
 		signer: AccountId,
 		who: Identity,
 		key: UserShieldingKeyType,
+		networks: Vec<Web3Network>,
 	) -> StfResult<UserShieldingKeyType> {
 		ensure!(
 			ensure_enclave_signer_or_self(&signer, who.to_account_id()),
 			StfError::SetUserShieldingKeyFailed(ErrorDetail::UnauthorizedSigner)
 		);
-		IMTCall::set_user_shielding_key { who, key }
+		IMTCall::set_user_shielding_key { who, key, networks }
 			.dispatch_bypass_filter(RuntimeOrigin::root())
 			.map_or_else(|e| Err(StfError::SetUserShieldingKeyFailed(e.error.into())), |_| Ok(key))
 	}
 
+	#[allow(clippy::too_many_arguments)]
+	pub fn handle_set_user_shielding_key<NodeMetadataRepository>(
+		calls: &mut Vec<OpaqueCall>,
+		node_metadata_repo: Arc<NodeMetadataRepository>,
+		signer: Identity,
+		who: Identity,
+		key: UserShieldingKeyType,
+		web3networks: Vec<Web3Network>,
+		hash: H256,
+		rpc_response_value: &mut Vec<u8>,
+	) -> StfResult<()>
+	where
+		NodeMetadataRepository: AccessNodeMetadata,
+		NodeMetadataRepository::MetadataType: NodeMetadataTrait,
+	{
+		debug!("set_user_shielding_key, who: {}", account_id_to_string(&who));
+		let account = SgxParentchainTypeConverter::convert(
+			who.to_account_id().ok_or(StfError::InvalidAccount)?,
+		);
+		let call_index =
```

Comment:
I think this point can be reached from two different calls but we are returning the same event for both, we should have two events as well

Reply:
Ok, I see there is no such call in parachain. Looks good then 👍🏼